app/components/groups/trash.resource.js

```diff
@@ -0,0 +1,22 @@
+;(function(){
+  angular
+    .module("todo")
+
+    .service("TrashResource", function($resource, API){
+     return $resource(API.BASE + API.TRASH + ":id/", {id: "@id"}, {
+        getGroups: {
+          method: "GET",
+          params: {id: null}
+        },
+        getGroup: {
+          method: "GET"
+        },
+        restoreGroup: {
+          method: "PATCH",
+        },
+        deleteGroup: {
```

Comment:
again, what the purpose of using custom names for crud operations? AFAIK, they might be called already https://docs.angularjs.org/api/ngResource/service/$resource